outgraphviz.c

```diff
@@ -128,8 +128,6 @@ int outnode_gra(ttreenode_t *pnode, treeparam_t *pparam)
 						 "labeljust=\"l\"; ",
 					sclustername, sclusterlabel);
 			}
-			free(sclustername);
-			free(sclusterlabel);
```

Comment:
Are you sure this is the right fix? Have you verified it under valgrind or with `-fsanitize=address`? The way the error handling is done in this function makes me really nervous.

I feel like we should clean up the control flow and error paths so it's more obvious what memory should be freed when.

Reply:
You are right, that removed the segmentation faults but lead to alog of memory leakage. I am now building with ASAN and try to fix memory leaks (also in other functions).

Even though cleaning up the control path did improve the situation, functions like slibcpy confuse my understanding of memory handling.

Reply:
Yeah, I'm not sure the original author had a good understanding of it either :sweat_smile:

Reply:
For now I removed the slibcpy function since it was a confusing wrapper for strdup. Before I am doing further cleanup I will push the changes and you can take another look at it.